### PR TITLE
Reopen #14418

### PR DESCRIPTION
**Pull request by igor-makarov**
Reopen #14418

### CODE REVIEW

File: Scripts/create_pods_and_versions_index.rb [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
-Dir['Specs/*/*/*/*/*'].each do |dir|
-  dir_components = dir.split('/')
-  pod, version = dir_components[-2, 2]
-  shard = dir_components[1..-3].join('_')
-  shards[shard] ||= {}
-  shards[shard][pod] ||= []
-  shards[shard][pod] << version
+source = Pod::Source.new('.')
+pods = source.pods # .first(10)
```
Comment (dnkoutso) [inline Scripts/create_pods_and_versions_index.rb]: nit remove `# .first(10)`?

File: Scripts/create_pods_and_versions_index.rb [hunk 2]
```diff
@@ -19,9 +22,27 @@
     end
   end
 end
+STDERR.puts "Generated #{shards.count} shards"
```
Comment (dnkoutso) [inline Scripts/create_pods_and_versions_index.rb]: why `STDERR`? was/is this used for debugging?

**Comment by dnkoutso** [comment]
LGTM. Please get one more approval perhaps by @orta to land this.

File: Scripts/create_pods_and_versions_index.rb (hunk 2 above)
Comment (igor-makarov) [reply Scripts/create_pods_and_versions_index.rb]: This is output that's displayed in the Netlify build logs. I think it's nice to keep it there.

**Comment by igor-makarov** [comment]
Picked the nit.

**Comment by orta** [comment]
Heyo, yep - this looks good to me. Let's give it a shot!

**Comment by igor-makarov** [comment]
On a side note, I'm beginning to think we need to split the builder code into a separate repo and have tests for it. But that's for if and when more changes are necessary.

**Pull request merged**